tests/ExpressionParserTest.php

```diff
@@ -411,6 +422,111 @@ public function testUnknownTestWithoutSuggestions()
         $parser->parse($env->tokenize(new Source('{{ 1 is foobar }}', 'index')));
     }
 
+    public function testNotReadyFunctionWithNoConstructor()
+    {
+        $env = new Environment(new ArrayLoader(), ['cache' => false, 'autoescape' => false]);
+        $env->addFunction(new TwigFunction('foo', 'foo', ['node_class' => NotReadyFunctionExpressionWithNoConstructor::class]));
+        $parser = new Parser($env);
+
+        $parser->parse($env->tokenize(new Source('{{ foo() }}', 'index')));
+        $this->doesNotPerformAssertions();
+    }
+
+    public function testNotReadyFilterWithNoConstructor()
+    {
+        $env = new Environment(new ArrayLoader(), ['cache' => false, 'autoescape' => false]);
+        $env->addFilter(new TwigFilter('foo', 'foo', ['node_class' => NotReadyFilterExpressionWithNoConstructor::class]));
+        $parser = new Parser($env);
+
+        $parser->parse($env->tokenize(new Source('{{ 1|foo }}', 'index')));
+        $this->doesNotPerformAssertions();
+    }
+
+    public function testNotReadyTestWithNoConstructor()
+    {
+        $env = new Environment(new ArrayLoader(), ['cache' => false, 'autoescape' => false]);
+        $env->addTest(new TwigTest('foo', 'foo', ['node_class' => NotReadyTestExpressionWithNoConstructor::class]));
+        $parser = new Parser($env);
+
+        $parser->parse($env->tokenize(new Source('{{ 1 is foo }}', 'index')));
+        $this->doesNotPerformAssertions();
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testNotReadyFunction()
+    {
+        $this->expectDeprecation('Since twig/twig 3.12: Twig node "Twig\Tests\NotReadyFunctionExpression" is not marked as ready for passing a "TwigFunction" instead instead of its name; please make it ready and then flag its constructor with the #[FirstClassTwigCallableReady] attribute.');
```

Comment:
```suggestion
        $this->expectDeprecation('Since twig/twig 3.12: Twig node "Twig\Tests\NotReadyFunctionExpression" is not marked as ready for passing a "TwigFunction" instead of its name; please make it ready and then flag its constructor with the #[FirstClassTwigCallableReady] attribute.');
```